internal/host/host.go

```diff
@@ -974,45 +977,60 @@ func (m *Manager) updateValidationsInDB(ctx context.Context, db *gorm.DB, h *mod
 	return hostutil.UpdateHost(logutil.FromContext(ctx, m.log), db, h.InfraEnvID, *h.ID, *h.Status, "validations_info", string(b))
 }
 
-func (m *Manager) AutoAssignRole(ctx context.Context, h *models.Host, db *gorm.DB) error {
-	// select role if needed
+func (m *Manager) AutoAssignRole(ctx context.Context, h *models.Host, db *gorm.DB) (bool, error) {
 	if h.Role == models.HostRoleAutoAssign {
-		return m.autoRoleSelection(ctx, h, db)
+		log := logutil.FromContext(ctx, m.log)
+		// If role is auto-assigned calculate the suggested roles
+		// This logic will moved to the monitor soon
+		suggestedRole, err := m.autoRoleSelection(ctx, h, db)
+		if err != nil {
+			return false, err
+		}
+
+		//copy the suggested role into the role and update the host record
+		log.Infof("suggested role %s for host %s cluster %s", suggestedRole, h.ID.String(), h.ClusterID.String())
+		if err := updateRole(m.log, h, suggestedRole, suggestedRole, db, swag.String(string(models.HostRoleAutoAssign))); err != nil {
+			log.WithError(err).Errorf("failed to update role %s for host %s cluster %s",
+				suggestedRole, h.ID.String(), h.ClusterID.String())
+			return true, err
```

Comment:
Shouldn't this return false?

Reply:
Computation was done even though it was not written.